public/app/features/dashboard-scene/scene/DashboardLinksControls.tsx

```diff
@@ -12,18 +15,39 @@ import { LINK_ICON_MAP } from '../settings/links/utils';
 
 export interface Props {
   links: DashboardLink[];
+  timeRange?: SceneTimeRangeLike;
   uid?: string;
 }
 
-export function DashboardLinksControls({ links, uid }: Props) {
+interface LinkInfo {
+  href: string;
+  title: string;
+  tooltip: string;
+}
+
+export function DashboardLinksControls({ links, uid, timeRange }: Props) {
+  const timeRangeState = timeRange?.useState();
```

Comment:
Should timeRange be optional? Not sure if this scenario could happen, but if a timeRange was passed, and then later was undefined, I think React would complain about inconsistent hook usage

Reply:
yes, it should. for this component at least

Reply:
Fixed, thank you!